gui/NumeReWindow.cpp

```diff
@@ -1917,6 +1917,11 @@ void NumeReWindow::OnMenuEvent(wxCommandEvent &event)
                     return;
             }
 
+            if(m_book->getCurrentEditor()->HasBeenSaved() && m_book->getCurrentEditor()->getFileType() == FILE_NPRC || m_book->getCurrentEditor()->getFileType() == FILE_NLYT || m_book->getCurrentEditor()->getFileType() == FILE_NSCR)
+            {
+                m_terminal->UpdateLibrary();
+            }

```

Comment:
```suggestion
  
            if (m_book->getCurrentEditor()->getFileType() == FILE_NPRC 
                || m_book->getCurrentEditor()->getFileType() == FILE_NLYT 
                || m_book->getCurrentEditor()->getFileType() == FILE_NSCR)
            {
                m_terminal->UpdateLibrary();
            }
          }
```
There is a logical problem and the necessary code is only needed, if the code is saved right now, not always, because in all other cases, the internal library will be refreshed after a few seconds idle time.